hack/generate-ta-tasks.sh

```diff
@@ -1,29 +1,23 @@
 #!/usr/bin/env bash
 
 set -o errexit
+set -o errtrace
 set -o nounset
 set -o pipefail
 set -o posix
 
 shopt -s globstar nullglob
 
+command -v go &> /dev/null || { echo Please install golang to run this tool; exit 1; }
+[[ "$(go env GOVERSION)" == @(go1|go1.[1-9]+(|.*|rc*|beta*)|go1.1[0-9]+(|.*|rc*|beta*)|go1.20*) ]] && { echo Please install golang 1.21.0 or newer; exit 1; }
```

Comment:
On my Fedora, by default I get
```
go: go.mod requires go >= 1.22.3 (running go 1.21.12; GOTOOLCHAIN=local)
```

Now, I happen to know that I need to run the script like this:

```bash
GOTOOLCHAIN=auto GOSUMDB=sum.golang.org hack/generate-ta-tasks.sh
```

But perhaps the script could try to detect this problem and advise what to do?

Reply:
I think I might have `GOTOOLCHAIN` turned on by default, I'll update the script to be a bit more helpful. I could also set `GOTOOLCHAIN=auto` in the script?

Reply:
Setting the vars in the script sounds okay. We'd need both GOTOOLCHAIN and GOSUMDB, otherwise I get

```
go: golang.org/toolchain@v0.0.1-go1.22.3.linux-amd64: verifying module: checksum database disabled by GOSUMDB=off
```

I think the Fedora default is GOTOOLCHAIN=local, GOSUMBD=off (at least I don't override them in my shell rcs)